pihole

```diff
@@ -96,6 +96,53 @@ versionFunc() {
     exit 0
 }
 
+restartDNS() {
+  dnsmasqPid=$(pidof dnsmasq)
+  if [[ ${dnsmasqPid} ]]; then
+    # service already running - reload config
+    if [ -x "$(command -v systemctl)" ]; then
+      systemctl restart dnsmasq
+    else
+      service dnsmasq restart
+    fi
+  else
+    # service not running, start it up
+    if [ -x "$(command -v systemctl)" ]; then
+      systemctl start dnsmasq
+    else
+      service dnsmasq start
+    fi
+  fi
+}
+
+piholeEnable() {
+  if [[ "${1}" == "0" ]] ; then
+    #Disable Pihole
+    sed -i 's/^addn-hosts/#addn-hosts/' /etc/dnsmasq.d/01-pihole.conf
+  else
+    #Enable pihole
+    sed -i 's/^#addn-hosts/addn-hosts/' /etc/dnsmasq.d/01-pihole.conf
+  fi
+  restartDNS
+}
+
+piholeStatus() {
+  if [[ $(cat /etc/dnsmasq.d/01-pihole.conf | grep "#addn-hosts=") ]] ; then
+    if [[ "${1}" == "web" ]] ; then echo 0; else echo ":::  pihole is Diabled"; fi
+  else
+    if [[ $(cat /etc/dnsmasq.d/01-pihole.conf | grep "addn-hosts=") ]] ; then
+      if [[ "${1}" == "web" ]] ; then echo 1; else echo ":::  pihole is Enabled"; fi
+    else
+      if [[ "${1}" == "web" ]] ; then echo 1; else
+        echo ":::  no hosts file linked to dnsmasq, adding it in enabled state"
+        echo "addn-hosts=/etc/pihole/gravity.list" >> /etc/dnsmasq.d/01-pihole.conf
+        restartDNS
+      fi
+    fi
```

Comment:
This might look better as:
```
if [[ "${1}" == "web" ]] ; then echo 1; else
  if [[ $(cat /etc/dnsmasq.d/01-pihole.conf | grep "addn-hosts=") ]] ; then
    echo ":::  pihole is Enabled";
  else
    echo ":::  no hosts file linked to dnsmasq, adding it in enabled state"
    echo "addn-hosts=/etc/pihole/gravity.list" >> /etc/dnsmasq.d/01-pihole.conf
    restartDNS
  fi
fi
```